elastic/kibana
# [Security Solution][Endpoint][Response Actions] Restrict actions log API requests via RBAC controls

## Summary

**Control access to Response Actions History page via Kibana RBAC privileges**

❗❗**MERGE AFTER**
❗❗elastic/kibana/pull/143366

- [x] Restricts access to response actions log API requests based on RBAC controls.
- [x] Restricts access to response actions history on endpoint details
- [x] Restricts access to response actions history on the response console  

**To test**
1. Create a role with `Security` `All` over all spaces, and Actions Log Management `Read` or `All`.
![Screenshot 2022-10-10 at 09 11 33](https://user-images.githubusercontent.com/1849116/194816301-004e268d-f661-4e7c-816e-d98c106cc843.png)
2. Create a user and assign it this new custom role with read access to `Actions Log Management`
3. Login with this user and access `app/security/administrations/response_actions_history`.
4. You should be able to view the page and notice that the API request is made in the `Network` tab of dev console.
5. Now edit the role and set the Actions Log Mangement permission to `None` and update the role.
6. Once you have updated the role, you should see a `Page not found` when you try and access `app/security/administrations/response_actions_history`. Also, you should notice that the side nav menu under `Managed` doesn't have a `Response actions history` item. 
7. Also, notice in the Network tab, there are no requests for the API. If you try and access the API manually then you should see a 403 error with a `Endpoint authorization failure` message.
8. Additionally you should see that the Endpoint details flyout does not show the `response actions history` tab anymore. And trying to force load the response actions history via URL params (`app/security/administrations/endpoints?selected_endpoint=<agentId>&show=activity_log`) still shows the `Overview` tab.

**endpoint details**
![response-actions-history-rbac](https://user-images.githubusercontent.com/1849116/194360961-fc122549-15f2-4ad0-8cee-870a0a76b9a8.gif)


### Checklist

### For maintainers
